src/app/dashboard-widgets/create-work-item-widget/create-work-item-widget.component.ts

```diff
@@ -14,41 +14,73 @@ import { filterOutClosedItems, WorkItemsData } from '../../shared/workitem-utils
 export class CreateWorkItemWidgetComponent implements OnInit {
 
   @Input() userOwnsSpace: boolean;
+  @Input() loggedInUser: User;
+  @Input() currentSpace: Space;
 
-  private _myWorkItems: ConnectableObservable<WorkItem[]>;
-  myWorkItemsCount: Observable<number>;
+  private _myWorkItems: WorkItem[];
```

Comment:
I would rather these remain Observables so that the component reacts to the values coming in.

Reply:
@aptmac we do not need observables as this has been taken care of in `ngOnChanges()` that will call `getCurrentSpaceWorkItems()` if any of the input properties changes.

Reply:
@divyanshiGupta I agree with @aptmac.

Reply:
> @sahil143 I don't think so using observables for everything is a good solution. There are scenarios where observables might create unnecessary changes in values. If you look at this issue [openshiftio/openshift.io#4348](https://github.com/openshiftio/openshift.io/issues/4348), it was introduced because of using observables when not needed.

I don't recall it being "because" of observables, but rather the log out event changed the loggedInUser value to be `{}` because there was no logged in user. As a result, calling `.trim()` on it somewhere caused an error.

> Regarding _myWorkItems being an Observable, i think it is not necessary here since its value will only be changing when the input properties change which will be handled by change detection and managed by the lifecycle hook.
In my opinion the code earlier was unnecessarily complicated by using ConnectableObservable, publishReplay(1) and _myWorkItems.connect().

I agree the older implementation might be a bit complicated; I recently refactored some of our frequently used services to move away from that type of implementation. Having said that, it needs to be replaced with something similar, and the work items come in as an observable it should be easy enough to keep it that way and let the front end update once the request finishes.

My main concern here is the assumption that subscribing to these observables completes before we need the data. We currently have (& had) some racy issues with services that provide absolute values of observables because of assumptions that it would be ready in time to serve the information. (IIRC, it was the user service providing both a `User` and `Observable<User>` value, and using the former could cause complications if it hadn't returned from the subscription fast enough .. e.g., in the case of slow internet connections).

Reply:
@aptmac I understand the concern for racy conditions. But, converting `_myWorkItems` as Observable will also need `myWorkItemCount` to be converted to an Observable since `myWorkItemCount` is being used in the template before `_myWorkItems` subscription using `async`. 
Something like this `this.myWorkItemsCount = this._myWorkItems.pipe(map(workItems => workItems.length));`.

Because of this there will be 4 subscriptions and API calls via `async` pipes instead of 1. That is my concern.

Reply:
What we can do here is add something like `*ngIf="(myWorkItems | async)?.length as myWorkItemsCount"` to one of the higher level divs in the html, and then reference it the count elsewhere in the html so the subscription only occurs once.

Reply:
@aptmac There are three places where we need to use `myWorkItemsCount`, two of them are with `ngIf` and first one is just `interpolation`. And they are distributed across multiple divs. Only higher level div left would be the top level div. If we add `*ngIf` to it like you suggested then until the workitem data returns from API call, the `div` will not be rendered at all which is not desired UX behaviour. 
the other way we can do this is to do a subscription for `myWorkItemsCount` in `component.ts` which will reduce the subscription to only 2. But then again it will defeat the whole thing of making it an Observable and using with `async` pipe in the first place.

Reply:
Also, about the racy conditions i don't think here it will be a problem since we are only rendering data in template and not trying to manipulate it anywhere.

Reply:
> There are three places where we need to use myWorkItemsCount, two of them are with ngIf and first one is just interpolation. And they are distributed across multiple divs. Only higher level div left would be the top level div. If we add *ngIf to it like you suggested then until the workitem data returns from API call, the div will not be rendered at all which is not desired UX behaviour.

We dont have to use `*ngIf`. We can use NgLetDirective(`*ngLet = "(myWorkItems | async) as workitems"`) from fabric8-planner. It can be used at top level div, it just store the results in a variable and then do `workitems.length` to get the workitem count